middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -128,47 +129,60 @@ let make_block ~dbg kind alloc_mode args =
 let block_load ~dbg (kind : P.Block_access_kind.t) (mutability : Mutability.t)
     ~block ~field =
   let mutability = Mutability.to_asttypes mutability in
-  let index = C.const ~dbg (Reg_width_const.tagged_immediate field) in
-  let load_func =
+  let field = Targetint_31_63.to_int field in
+  let load_func, offset =
     match kind with
     | Mixed { field_kind = Value_prefix Any_value; _ }
     | Values { field_kind = Any_value; _ } ->
-      C.get_field_computed Pointer
+      C.get_field_computed Pointer, field
     | Mixed { field_kind = Value_prefix Immediate; _ }
     | Values { field_kind = Immediate; _ } ->
-      C.get_field_computed Immediate
-    | Naked_floats _ -> C.unboxed_float_array_ref
-    | Mixed { field_kind = Flat_suffix field_kind; _ } -> (
-      match field_kind with
-      | Tagged_immediate -> C.get_field_computed Immediate
-      | Naked_float -> C.unboxed_float_array_ref
-      | Naked_float32 -> C.get_field_unboxed_float32
-      | Naked_int32 -> C.get_field_unboxed_int32
-      | Naked_int64 | Naked_nativeint -> C.get_field_unboxed_int64_or_nativeint)
+      C.get_field_computed Immediate, field
+    | Naked_floats _ -> C.unboxed_float_array_ref, field
+    | Mixed { field_kind = Flat_suffix field_kind; shape; _ } ->
+      let func =
+        match field_kind with
+        | Tagged_immediate -> C.get_field_computed Immediate
+        | Naked_float -> C.unboxed_float_array_ref
```

Comment:
Please leave a CR to make the naming of this function consistent, and likewise for "setfield_computed" below (should be "set_field_computed").